src/ts/settlement.ts

```diff
@@ -147,8 +154,21 @@ export const FLAG_MASKS = {
     offset: 1,
     options: [false, true],
   },
-  signingScheme: {
+  sellTokenBalance: {
     offset: 2,
+    options: [
+      OrderBalance.ERC20,
+      undefined, // unused
+      OrderBalance.EXTERNAL,
+      OrderBalance.INTERNAL,
+    ],
```

Comment:
> I like how undefined is used to denote an unused flag, but technically the second slot is also OrderBalance.ERC20 according to the smart contract.

Correct.

> What if we use OrderBalance.ERC20 in the second slot, and we change the decoding test to use a new allEncodedUniqueFlagOptions instead of the current allEncodedFlagOptions?

I like this idea a lot.

Reply:
https://github.com/gnosis/gp-v2-contracts/issues/614